blockchain/src/history/history_store.rs

```diff
@@ -44,12 +44,13 @@ pub struct HistoryStore {
     /// A database of all historic transactions indexed by their hash (= leaf hash in the history
     /// tree).
     hist_tx_table: TableProxy,
-    /// A database of all leaf hashes and indexes indexed by the hash of the transaction. This way we
-    /// can start with a transaction hash and find it in the MMR.
+    /// A database of all leaf hashes and indexes indexed by the hash of the (raw) transaction. This way we
+    /// can start with a raw transaction hash and find it in the MMR.
+    /// Mapping of raw tx to leaf hash  (= executed tx hash + index).
```

Comment:
There seems to be a mismatch between here and what we actually put in (as it sometimes seems to be the ext. tx hash).

Reply:
The executed tx hash and the raw tx hash are the same for these cases. it only defers on the HistoricTransactionData::Basic(tx).